### PR TITLE
shutdown when tab closed Fixes #737

### LINKED ISSUE #737: Firefox OS Simulator doesn't close when Dashboard page is closed.
### Steps to reproduce:
1. Open resource://r2d2b2g-at-mozilla-dot-org/r2d2b2g/data/content/index.html or 
   Tools > WebDeveloper > Firefox OS Simulator.
2. Once the simulator starts, open the Amazon app.
3. Close the Dashboard screen on Firefox/Aurora
4. The Simulator doesn't quit. Daemon still keeps running on port.
### Required effect:

The simulator should close too.

### PR DESCRIPTION
**Pull request by nickdesaulniers**
review? @ochameau

### CODE REVIEW

File: addon/lib/main.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ PageMod({
   contentScriptFile: Simulator.contentScript,
   contentScriptWhen: 'start',
   onAttach: function(worker) {
+    worker.on('detach', Simulator.unload.bind(Simulator));
```
Comment (ochameau) [inline addon/lib/main.js]: I'd rather call Simulator unload from `Simulator.worker` setter, where we already listen for `detach`.

**Comment by ochameau** [comment]
Sorry for not seeing the request sooner, I'm really bad at tracking github notifications...

**Comment by nickdesaulniers** [comment]
review? @ochameau

**Comment by ochameau** [comment]
Looks good, thanks!

**Pull request merged**